route/route_test.go

```diff
@@ -612,3 +613,109 @@ func TestGRPCHealthProbeWatch(t *testing.T) {
 	sentMessage := mockServer.GetSentMessages()[0]
 	assert.Equal(t, grpc_health_v1.HealthCheckResponse_SERVING, sentMessage.Status)
 }
+
+func TestGetDatasetFromRequest(t *testing.T) {
+	testCases := []struct {
+		name                string
+		datasetName         string
+		expectedDatasetName string
+		expectedError       error
+	}{
+		{
+			name:          "empty dataset name",
+			datasetName:   "",
+			expectedError: fmt.Errorf("missing dataset name"),
+		},
+		{
+			name:          "dataset name with invalid URL encoding",
+			datasetName:   "foo%2",
+			expectedError: url.EscapeError("%2"),
+		},
+		{
+			name:                "normal dataset name",
+			datasetName:         "foo",
+			expectedDatasetName: "foo",
+		},
+		{
+			name:                "dataset name with numbers",
+			datasetName:         "foo123",
+			expectedDatasetName: "foo123",
+		},
+		{
+			name:                "dataset name with hyphen",
+			datasetName:         "foo-bar",
+			expectedDatasetName: "foo-bar",
+		},
+		{
+			name:                "dataset name with underscore",
+			datasetName:         "foo_bar",
+			expectedDatasetName: "foo_bar",
+		},
+		{
+			name:                "dataset name with tilde",
+			datasetName:         "foo~bar",
+			expectedDatasetName: "foo~bar",
+		},
+		{
+			name:                "dataset name with period",
+			datasetName:         "foo.bar",
+			expectedDatasetName: "foo.bar",
+		},
+		{
+			name:                "dataset name with URL encoded hyphen",
+			datasetName:         "foo%2Dbar",
+			expectedDatasetName: "foo-bar",
+		},
+		{
+			name:                "dataset name with URL encoded underscore",
+			datasetName:         "foo%5Fbar",
+			expectedDatasetName: "foo_bar",
+		},
+		{
+			name:                "dataset name with URL encoded tilde",
+			datasetName:         "foo%7Ebar",
+			expectedDatasetName: "foo~bar",
+		},
+		{
+			name:                "dataset name with URL encoded period",
```

Comment:
just for completeness, how about a dataset name with a `%` symbol?

Reply:
Added extra tests in https://github.com/honeycombio/refinery/pull/1199/commits/492e57250ae9b58d5e18d64c93ac24187755bf3b